### PR TITLE
Feature/handle errors

### PR DESCRIPTION
**Pull request by Sh3lldor**
# Handle errors

1) Added err model for every js.
2) Pop up err model for every ajax error.
3) Added header for personal/admin panels.

## How to

1) Set <b>errModal</b> in the top of the JS script.
2) Added the following script under err
```
    $('body').append(errModal);
    $(".modal-title").html("Error")
    $(".modal-body").append("<p>" + err["responseJSON"].error + "</p>")
    $("#statusModal").modal('show');
```

### CODE REVIEW

**Comment by Sh3lldor** [comment]
Waiting for @DianaAlankry and @YuvalYakar to merge their branch first, I touched some of their stuff of this branch (header in personal page).

First merge `feature/personal-page` and then merge `feature/handle errors` branch.


From here, I am starting to add more features to admin panel.
1) Add, update and delete rooms via text-box or input json file.
2) Add new users
3) TDL: Add services modal and connect services from admin panel to DB.

**Comment by Sh3lldor** [comment]
# Whats new
1) Added dynamic services and rooms
2) Ordered rooms are now by DB rooms.
3) Admin can add new rooms with config file or by text box
4) Admin can add new services by name and cost

## Review
@AmielRe please review this 🥇

File: config/response.js [hunk 1]
```diff
@@ -2,6 +2,33 @@ const response = {
     "status" : {
         "200" : "OK !",
         "500" : "Something has happend"
+    },
```
Comment (AmielRe) [inline config/response.js:5]: This is great!

File: controllers/chatController.js [hunk 2]
```diff
@@ -42,7 +43,7 @@ const getAllUserMessages = (req, res) => {
         
         function(err, chats) {
         if (err) {
-            res.status(500).json({"status": "Error getting chats."})
+            res.status(500).json({"error":Response.chat.queryError})
```
Comment (AmielRe) [inline controllers/chatController.js]:
```suggestion
            res.status(500).json({"error": Response.chat.queryError})
```

File: controllers/ordersController.js [hunk 3]
```diff
@@ -81,15 +96,19 @@ const deleteOrder = (req,res) => {
 const parseRooms = (req) => {
     const roomTypes = req.body.roomType;
     const roomPrices = req.body.roomPrice;
+    const roomIds = req.body.roomIds;
+    console.log(roomIds);
```
Comment (AmielRe) [inline controllers/ordersController.js]:
```suggestion

```

File: controllers/ordersController.js [hunk 4]
```diff
@@ -81,15 +96,19 @@ const deleteOrder = (req,res) => {
...
     let totalCost = 0;
 
     for (let i = 0; i < roomTypes.length; i++) {
-        const room = new Room({
+        const room = Room({
+            _id: roomIds[i],
             roomType: roomTypes[i],
             cost: parseInt(roomPrices[i])
         });
+        console.log(room);
```
Comment (AmielRe) [inline controllers/ordersController.js]:
```suggestion
```

File: controllers/usersController.js [hunk 5]
```diff
@@ -20,18 +21,21 @@ const getAllAdmins = async (req,res) => {
...
+        res.status(500).json({"error": Response.user.queryError});
     }
 }
 
 // Register new user to the system
 const addNewUser = async (req,res) => {
 
+    // Gets the jwt token, if he is an admin -> true else false
+    isAdmin = getJwtDetails(req.cookies.jwt)["role"] == Config.ROLES.admin ? true : false;
+    console.log(isAdmin);
```
Comment (AmielRe) [inline controllers/usersController.js]:
```suggestion
```

File: public/js/admin/ratings.js [hunk 6]
```diff
@@ -36,7 +54,10 @@ $(document).ready(function(){
       setGraph()
     },
     error: function(err){
-        
+      $('body').append(errModal);
```
Comment (AmielRe) [inline public/js/admin/ratings.js:57]: Maybe export this to a function to avoid the double code?

File: public/js/admin/admin-panel.js [hunk 7]
```diff
@@ -144,7 +164,10 @@ function getRooms(fill_table=false) {
             }
         },
         error: function(err){
-            console.log(err)
+            $('body').append(errModal);
```
Comment (AmielRe) [inline public/js/admin/admin-panel.js:167]: Maybe export this to a function to avoid double code?

File: public/js/admin/roles.js [hunk 8]
```diff
@@ -16,7 +34,10 @@ $(document).ready(function (){
               });
         },
         error: function(err){
-            console.log(err)
+            $('body').append(errModal);
```
Comment (AmielRe) [inline public/js/admin/roles.js:37]: Maybe export this to a function to avoid the double code?

File: public/js/chats/chats-admins.js [hunk 9]
```diff
@@ -66,7 +84,10 @@ function getMsg(email) {
             };
         },
         error: function(err){
-            console.log(err)
+            $('body').append(errModal);
```
Comment (AmielRe) [inline public/js/chats/chats-admins.js:87]: Maybe export this to a function to avoid the double code?

File: public/js/chats/chats-guests.js [hunk 10]
```diff
@@ -65,7 +83,10 @@ function getMsg(email) {
             };
         },
         error: function(err){
-            console.log(err)
+            $('body').append(errModal);
```
Comment (AmielRe) [inline public/js/chats/chats-guests.js:86]: Maybe export this to a function to avoid the double code?

File: public/js/confirmation.js [hunk 11]
```diff
@@ -67,6 +88,10 @@ $(() => {
             },
             error: function(err){
                 console.log(err)
+                //$('body').append(errModal);
```
Comment (AmielRe) [inline public/js/confirmation.js:91]: Remove?

File: public/js/login.js [hunk 12]
```diff
@@ -16,8 +16,8 @@ $("#basic-auth-form").submit(function(e) {
             document.location.href = response.redirect;
         },
         error: function(err){
-            $(".modal-title").html("Error")
-            $(".modal-body").append("<p>" + err["responseJSON"].error + "</p>")
+            $(".modal-title-status").html("Error")
```
Comment (AmielRe) [inline public/js/login.js:19]: Maybe export this to a function to avoid the double code?

File: views/chats.ejs [hunk 13]
```diff
@@ -85,7 +64,7 @@
     <%- include('partials/footer'); %>
   </footer>
   <%- include('partials/scripts'); %>
-  <% if (isAdmin) { %>
+  <% if ( jwt.role == 1 ) { %>
```
Comment (AmielRe) [inline views/chats.ejs:69]: Maybe add a comment saying 1 equals admin?

File: views/reservation.ejs [hunk 14]
```diff
@@ -107,175 +90,16 @@
...
-                </div>
-                <div class="col" data-price="60" data-col-type="col-standard" data-display-name="Standard">
-                  
-                </div>
-                <div class="col" data-price="60" data-col-type="col-standard" data-display-name="Standard">
-                  
-                </div>
-                <div class="col" data-price="60" data-col-type="col-standard" data-display-name="Standard">
-                  
+                  <div class="p-3 border bg-light text-center rounded fw-bold">&nbsp;Food<br>Court 2<br></div>
```
Comment (AmielRe) [inline views/reservation.ejs:100]: Remove?

**Review by AmielRe — COMMENTED** [review]
Great job, loved it!

**Pull request merged**